test/commands/extended/is_promotable_test.py

```diff
@@ -0,0 +1,407 @@
+
+    def test_happy_path(self):
+        """
+        Successfully checking promotability.
+        """
+        # To pass timestamp check, we need a timestamp that defines
+        # a moment no earlier than 660s from the moment the function
+        # is called during test. To do this, we cheat a bit:
+        tx = Transaction.from_tryte_string(self.trytes1)
+        tx.attachment_timestamp = get_current_ms()
+        # self.trytes1 becomes invalid transaction since we modified
+        # attachment_timestamp, but it doesn't matter here
+        self.trytes1 = tx.as_tryte_string()
+        # The same for self.trytes2
+        tx = Transaction.from_tryte_string(self.trytes2)
+        tx.attachment_timestamp = get_current_ms()
+        self.trytes2 = tx.as_tryte_string()
+
+        self.adapter.seed_response('checkConsistency', {
+            'state': True,
+        })
+        self.adapter.seed_response('getTrytes', {
+            'trytes': [self.trytes1, self.trytes2]
+        })
+        
+        response = self.command(tails=[self.hash1, self.hash2])
+
+        self.assertDictEqual(
+            response,
+
+            {
+                'promotable': True,
+            }
+        )
+
+    def test_not_consistent(self):
+        """
+        One of the tails is not consistent.
+        """
+
+        self.adapter.seed_response('checkConsistency', {
+            'state': False,
+            'info': 'Oops, something went wrong.',
+        })
+
+        # No need for mokcing `getTrytes` becasue we should not
+        # reach that part
+
+        response = self.command(tails=[self.hash1, self.hash2])
+
+        self.assertDictEqual(
+            response,
+
+            {
+                'promotable': False,
+                'info': 'Oops, something went wrong.',
+            }
+        )
+    
+    def test_one_timestamp_invalid(self):
+        """
+        Test invalid timestamp in one of the transactions.
+        """
+        # Note that self.trytes2 will have the original and
+        # therefore invalid (too old) timestamp
+        tx = Transaction.from_tryte_string(self.trytes1)
+        tx.attachment_timestamp = get_current_ms()
+        self.trytes1 = tx.as_tryte_string()
+
+        self.adapter.seed_response('checkConsistency', {
+            'state': True,
+        })
+        self.adapter.seed_response('getTrytes', {
+            'trytes': [self.trytes1, self.trytes2]
+        })
+
+        response = self.command(tails=[self.hash1, self.hash2])
+
+        self.assertDictEqual(
+            response,
+
+            {
+                'promotable': False,
+                'info': ['Transaction {tx_hash} is above max depth.'.format(
+                    tx_hash=self.hash2
+                )],
+            }
+        )
+
+    def test_is_within_depth(self):
+        """
+        Test ``is_within_depth`` helper method.
+        """
+        # Timestamp is too old (depth=3)
+        old_timestamp = get_current_ms() - 660000
```

Comment:
❔ Should there be a reference to `MILESTONE_INTERVAL` here, instead of a magic number?

Reply:
yep